src/swap_if_arms.rs

```diff
@@ -0,0 +1,53 @@
+//! lint on if branches that could be swapped so no `!` operation is necessary on the condition
+
+use rustc::lint::*;
+use syntax::attr::*;
+use syntax::ast::*;
+
+use utils::span_help_and_lint;
+
+/// **What it does:** Warns on enum variants that are prefixed or suffixed by the same characters
+///
+/// **Why is this bad?** Enum variant names should specify their variant, not the enum, too.
+///
+/// **Known problems:** None
+///
+/// **Example:** enum Cake { BlackForestCake, HummingbirdCake }
+declare_lint! {
+    pub SWAP_IF_ARMS, Warn,
```

Comment:
perhaps something more like IF_NOT_ELSE? Our lint names usually describe the cause, not the solution.